### PR TITLE
[Feature] Add 'validate' option

### PR DESCRIPTION
**Pull request by bacali95**
The purpose of this MR is to give the developer an extra option to extra validates the config value before its usage in the app.

### CODE REVIEW

File: src/ConfigManager.ts [hunk 1]
```diff
@@ -42,6 +48,9 @@ class ConfigManager {
         // Make sure that any value that is required, has actually been set by either a default value, a config file or env.
         validateRequiredConfigValues(configInstance, configClass);
 
+        // Run extra validations passed by the user
+        extraValidateConfigValues(configInstance, configClass);
```
Comment (guidsdo) [inline src/ConfigManager.ts]: Let's rename the name of the method to: "runCustomValidations" so we don't need the comment

File: src/types.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export type ConfigValueOptions = {
      */
     name: string;
     recommendedValue?: SupportedConfigTypes;
+    extraValidations?: (value: any) => void;
```
Comment (guidsdo) [inline src/types.ts]: Let's just call it `validate` and add a comment that says something like you can do simple non-async validations that will prevent the config from loading (and thus your program from running) when throwing.

Also; we can also let it return a boolean and when false just throw the method so the end-user can see what condition is been required. This way you can either throw a nice custom error message or just use a simply condition.

**Review by guidsdo — COMMENTED** [review]
Thank you for your contribution! I've left two comments, please have a look

File: src/types.ts (hunk 2 above)
Comment (bacali95) [reply src/types.ts]: I gave it try, please check again

File: src/ConfigManager.ts (hunk 1 above)
Comment (bacali95) [reply src/ConfigManager.ts]: Done.

**Review by guidsdo — APPROVED** [review]
Thanks!

**Pull request merged**